app/controllers/ProjectSidewalkAPIController.scala

```diff
@@ -164,34 +177,114 @@ class ProjectSidewalkAPIController @Inject()(implicit val env: Environment[User,
+    def prepareFeatureCollectionCSV = {
+      val labelsForScore: List[AttributeForAccessScore] = getLabelsForScore(version, coordinates)
+      val allStreetEdges: List[StreetEdge] = StreetEdgeTable.selectStreetsIntersecting(coordinates(0), coordinates(2), coordinates(1), coordinates(3))
+      val auditedStreetEdges: List[StreetEdge] = StreetEdgeTable.selectAuditedStreetsIntersecting(coordinates(0), coordinates(2), coordinates(1), coordinates(3))
+      val neighborhoods: List[NamedRegion] = RegionTable.selectNamedNeighborhoodsWithin(lat1, lng1, lat2, lng2)
+      val significance = Array(0.75, -1.0, -1.0, -1.0)
+      for(neighborhood <- neighborhoods) {
+        val coordinates: Array[Coordinate] = neighborhood.geom.getCoordinates
+        val auditedStreetsIntersectingTheNeighborhood = auditedStreetEdges.filter(_.geom.intersects(neighborhood.geom))
+        var coordinatesToString: String = "\"["
+        for(coordinate <- coordinates) {
+          coordinatesToString += "(" + coordinate.x + "," + coordinate.y + "),"
+        }
+        if(coordinates.length > 0) {  // correct formatting for commas
+          coordinatesToString = coordinatesToString.substring(0, coordinatesToString.length() - 1)
+        }
+        coordinatesToString += "]\""
+        if (auditedStreetsIntersectingTheNeighborhood.nonEmpty) {
+          val streetAccessScores: List[AccessScoreStreet] = computeAccessScoresForStreets(auditedStreetsIntersectingTheNeighborhood, labelsForScore)  // I'm just interested in getting the attributes
+          val averagedStreetFeatures = streetAccessScores.map(_.attributes).transpose.map(_.sum / streetAccessScores.size).toArray
+          val accessScore: Double = computeAccessScore(averagedStreetFeatures, significance)
 
+          val allStreetsIntersectingTheNeighborhood = allStreetEdges.filter(_.geom.intersects(neighborhood.geom))
+          val coverage: Double = auditedStreetsIntersectingTheNeighborhood.size.toDouble / allStreetsIntersectingTheNeighborhood.size
+
+          assert(coverage <= 1.0)
+
+          writer.println(neighborhood.name.getOrElse("NA") + "," + neighborhood.regionId + "," + accessScore + "," + coordinatesToString + "," + coverage + "," + averagedStreetFeatures(0) + "," + averagedStreetFeatures(1) + "," + averagedStreetFeatures(2) + "," + averagedStreetFeatures(3) + "," + 
+                        significance(0) + "," + significance(1) + "," + significance(2) + "," + significance(3))                
+        } else {
+          writer.println(neighborhood.name.getOrElse("NA") + "," + neighborhood.regionId + "," + None.asInstanceOf[Option[Double]] + "," + coordinatesToString + ",,,,,," + significance(0) + "," + significance(1) + "," + significance(2) + "," + significance(3))
```

Comment:
Instead of using `None.asInstanceOf[Option[Double]]` or leaving the cell blank, can you use "NA"?